lpc55_sign/src/bin/lpc55_sign.rs

```diff
@@ -135,80 +127,94 @@ fn main() -> Result<()> {
             address,
         } => {
             crc_image::update_crc(&src_bin, &dest_bin, address)?;
-            println!("Done! CRC image written to {:?}", &dest_bin);
+            check!(OK, "CRC image written to {:?}", &dest_bin);
         }
         Command::ChainedImage {
-            with_dice,
-            with_dice_inc_nxp_cfg,
-            with_dice_cust_cfg,
-            with_dice_inc_sec_epoch,
-            src_bin,
-            cfg,
-            dest_bin,
-            dest_cmpa,
-            address,
+            dice_args,
+            image_args:
+                ImageArgs {
+                    address,
+                    src_bin,
+                    dest_bin,
+                    dest_cmpa,
+                    dest_cfpa,
+                },
+            cert_cfg,
         } => {
-            let cfg_contents = std::fs::read(&cfg)?;
-            let toml: CfgFile = toml::from_slice(&cfg_contents)?;
-
-            let rkth = signed_image::sign_chain(&src_bin, None, &toml.certs, &dest_bin, address)?;
-            signed_image::create_cmpa(
-                with_dice,
-                with_dice_inc_nxp_cfg,
-                with_dice_cust_cfg,
-                with_dice_inc_sec_epoch,
-                &rkth,
-                &dest_cmpa,
-            )?;
-
-            println!(
-                "Done! Signed image written to {:?}, CMPA to {:?}",
-                &dest_bin, &dest_cmpa
-            );
+            let cfg_contents = std::fs::read(&cert_cfg)?;
+            let cfg: CertConfig = toml::from_slice(&cfg_contents)?;
+            let private_key = std::fs::read_to_string(&cfg.private_key)?;
+            let signing_certs = read_certs(&cfg.signing_certs)?;
+            let mut root_certs = read_certs(&cfg.root_certs)?;
+            if root_certs.len() < 4 {
+                let empty_roots = [vec![], vec![], vec![], vec![]];
+                root_certs.extend_from_slice(&empty_roots[..4 - root_certs.len()]);
+            } else if root_certs.len() > 4 {
+                bail!("Too many roots, max four");
+            }
```

Comment:
This is oddly phrased, how about
```suggestion
            if root_certs.len() > 4 {
                bail!("Too many roots, max four");
            }
            root_certs.resize_with(4, Vec::new);
```

Reply:
Fixed with `pad_roots`.